### DISCUSSION TITLE
Filter above table (in widget?)

### DISCUSSION THREAD

**Discussion by G3z** [filamentphp/filament#2217] [Help] [answered]
Is it possible to show filters before the table or use a custom component to filter the table ? 
I have a lot of filters (15) and i thinks it's not good ux to put them in the pop over menu.

**Comment by davidfrigolet** [comment, marked as answer]
Use Livewire events https://laravel-livewire.com/docs/2.x/events

In widget view 

```
<x-filament::widget>
    <x-filament::card>
        <button wire:click="$emit('test')">run event</button>
        
    </x-filament::card>
</x-filament::widget>
```

In ListRecords

```
    protected $listeners = ['test' => 'listenerMethod'];

    protected $queryString = [
        'tableFilters',
        'tableSortColumn',
        'tableSortDirection',
        'tableSearchQuery' => ['except' => ''],
    ];

    public function listenerMethod()
    {
        $this->tableFilters['MYFILTER']['value'] = 'value';
        $this->notify('success', 'event listened');
    }
```

**Reply by G3z** [reply]
thanks for your response!
just to be clear: listRecords is a custom livewire component that uses [Filament's Table](https://filamentphp.com/docs/2.x/tables/getting-started#preparing-your-livewire-component) and I should avoid using [resource's table](https://filamentphp.com/docs/2.x/admin/resources#tables) right ?

**Reply by danharrin** [reply]
The resource is just a DTO which passes the columns and filter's to the table package :) any core customizations can be made to the list page whilst still using the resource table schema
